xen/arch/arm/irq.c

```diff
@@ -191,12 +191,14 @@ int request_irq(unsigned int irq, unsigned int irqflags,
 void remove_gsx_guest(struct domain *d)
 {
     struct irq_desc *desc = irq_to_desc(gsx_irq_num);
-    struct irq_guest *info = irq_get_guest_info(desc);
+    struct irq_guest *info;
     unsigned long flags;
     int i;
 
     spin_lock_irqsave(&desc->lock, flags);
 
+    info = irq_get_guest_info(desc);
+
```

Comment:
I have no idea why this change help with mentioned assert. Could you please elaborate commit description?